home-application/src/main/java/com/softserveinc/ita/homeproject/application/api/PollApiImpl.java

```diff
@@ -78,68 +96,69 @@ public Response deletePolledHouse(Long pollId, Long id) {
         return Response.status(Response.Status.NO_CONTENT).build();
     }
 
-    @PreAuthorize(DELETE_QUESTION_PERMISSION)
+    @PreAuthorize(GET_POLL_HOUSE_PERMISSION)
     @Override
-    public Response deleteQuestion(Long pollId, Long id) {
-        pollQuestionService.deactivatePollQuestion(pollId, id);
-        return Response.status(Response.Status.NO_CONTENT).build();
+    public Response getPolledHouse(Long pollId, Long id) {
+        var toGet = houseService.getOne(id, getSpecification());
+        var readHouse = mapper.convert(toGet, ReadHouse.class);
+
+        return Response.status(Response.Status.OK).entity(readHouse).build();
     }
 
+
     @PreAuthorize(GET_POLL_PERMISSION)
     @Override
-    public Response getPoll(Long id) {
-        var pollDto = pollService.getOne(id);
-        var readPoll = mapper.convert(pollDto, ReadPoll.class);
-        return Response.status(Response.Status.OK).entity(readPoll).build();
+    public Response queryPoll(
+        Long cooperationId,
+        Integer pageNumber,
+        Integer pageSize,
+        String sort,
+        String filter,
+        Long id,
+        LocalDateTime creationDate,
+        LocalDateTime completionDate,
+        PollType type,
+        PollStatus status
+    ) {
+        Page<PollDto> readPoll = pollService.findAll(pageNumber, pageSize, getSpecification());
+        return buildQueryResponse(readPoll, ReadPoll.class);
     }
 
-    @PreAuthorize(GET_POLL_HOUSE_PERMISSION)
+    @PreAuthorize(CREATE_QUESTION_PERMISSION)
     @Override
-    public Response getPolledHouse(Long pollId, Long id) {
-        var toGet = houseService.getOne(id, getSpecification());
-        var readHouse = mapper.convert(toGet, ReadHouse.class);
+    public Response createQuestion(Long pollId, CreateQuestion createQuestion) {
+        var createQuestionDto = mapper.convert(createQuestion, PollQuestionDto.class);
+        var readQuestionDto = pollQuestionService.createPollQuestion(pollId, createQuestionDto);
+        var readQuestion = mapper.convert(readQuestionDto, ReadQuestion.class);
 
-        return Response.status(Response.Status.OK).entity(readHouse).build();
+        return Response.status(Response.Status.CREATED).entity(readQuestion).build();
     }
 
-    @PreAuthorize(GET_QUESTION_PERMISSION)
+    @PreAuthorize(CREATE_VOTE_PERMISSION)
     @Override
-    public Response getQuestion(Long pollId, Long id) {
-        PollQuestionDto toGet = pollQuestionService.getOne(id, getSpecification());
-        var readQuestion = mapper.convert(toGet, ReadMultipleChoiceQuestion.class);
-
-        return Response.status(Response.Status.OK).entity(readQuestion).build();
+    public Response createVote(Long pollId, CreateVote createVote) {
+        User currentUser = ((HomeUserWrapperDetails) userDetailsService.loadUserByUsername(
+            SecurityContextHolder.getContext().getAuthentication().getName())).getUser();
```

Comment:
You can use Spring annotation for it.
public Response method(@AuthenticationPrincipal HomeUserWrapperDetails user)
It will inject current user in method.

Reply:
@AuthenticationPrincipal has @Target ({ElementType.PARAMETER, ElementType.ANNOTATION_TYPE}). That is, to use it, you need to add the Username parameter to the method parameters. But our method is autogenerated, so I prefer to get User from SecurityContextHolder.